src/main/java/pathfinding/TextualDirections.java

```diff
@@ -4,6 +4,8 @@
 
 import java.util.ArrayList;
 
+import static ui.ProviderBox.database;
```

Comment:
This makes me very nervous. Why does TextualDirections need database access? If it uses this at all, it also breaks our ability to test textual directions independently.